cmd/zdb/zdb.c

```diff
@@ -9129,6 +9137,31 @@ main(int argc, char **argv)
 	if (argc < 2 && dump_opt['R'])
 		usage();
 
+	target = argv[0];
+
+	/*
+	 * Automate cachefile
+	 */
+	if (!spa_config_path_env && !config_path_console && target) {
+		zpool_handle_t *zhp;
+		char prop[ZFS_MAXPROPLEN];
+		zprop_source_t srctype = ZPROP_SRC_LOCAL;
+		if ((g_zfs = libzfs_init()) && (zhp = zpool_open_canfail(g_zfs,
+		    target)) && !zpool_get_prop(zhp, ZPOOL_PROP_CACHEFILE, prop,
+		    sizeof (prop), &srctype, B_FALSE) && prop[0] != '\0') {
```

Comment:
Parts like that I prefer to wrap on logical boundaries, like `&&` here. I think it would be more readable.

Also, since we are trying to free allocated memory in first patch, shouldn't somewhere here be called zpool_close()?